### DISCUSSION TITLE
What am I missing on API filter

### DISCUSSION THREAD

**Discussion by gojrs** [pocketbase/pocketbase#5681] [Q&A]
I am not sure where I can get a better understanding of what I am doing wrong. When I query from inside Go, I get results from a query. I try the same query from either the SDK or from Postman,  I get zero results. I am assuming it is related to the filter as if I remove it I get the "admin only" error (as expected)

Client SDK:
 (async () => {
                const result = await pb.collection('confGen_template').getList(1,20,{
                    filter : pb.filter("owner_id = {:owner}", { owner: "45xi5y1b7vphoz9"})
                })
                console.log(result)

Server query:
err := crs.db.Select("id", "owner_id", "template", "file_name").
		From("confGen_template").
		Where(dbx.NewExp("owner_id = {:owner}", dbx.Params{"owner": ownerId})).
		All(&items)

@request.auth.id != ""

my understanding is that this rule ( as a test) should require a token, or should be authenticated

**Comment by nerdlibfront** [comment]
In the Client SDK you need to be authenticated (your rule states that -> `@request.auth.id != ""`
You are probably not authenticating on client side. So you need to do some login first like using `pb.collection('users').authWithPassword(...)`
